sw/SoC/startup/startup.s

```diff
@@ -1,85 +1,73 @@
-# Copyright lowRISC contributors.
-# Licensed under the Apache License, Version 2.0, see LICENSE for details.
-# SPDX-License-Identifier: Apache-2.0
-#include "demo_system_regs.h"
+# UninaSoC startup file
 
-.section .text
+.section .vector_table, "ax"
+.option norvc;
 
+  # According to RISC-V Specification, all entries are jumps to the specific handler.
+  # Only the reset handler is defined in this file, while all other handlers points to
+  # the default_handler (a loop)
 
-reset_handler:
+  jal x0, _reset_handler         
+  .rept 31
+  jal x0, _default_handler
+  .endr        
 
-  /* set all registers to zero */
-  mv  x1, x0
-  mv  x2, x1
-  mv  x3, x1
-  mv  x4, x1
-  mv  x5, x1
-  mv  x6, x1
-  mv  x7, x1
-  mv  x8, x1
-  mv  x9, x1
-  mv x10, x1
-  mv x11, x1
-  mv x12, x1
-  mv x13, x1
-  mv x14, x1
-  mv x15, x1
-  mv x16, x1
-  mv x17, x1
-  mv x18, x1
-  mv x19, x1
-  mv x20, x1
-  mv x21, x1
-  mv x22, x1
-  mv x23, x1
-  mv x24, x1
-  mv x25, x1
-  mv x26, x1
-  mv x27, x1
-  mv x28, x1
-  mv x29, x1
-  mv x30, x1
-  mv x31, x1
+.section .text.handlers
 
-  /* stack initilization */
+_reset_handler:
+  .global _reset_handler
+
+  # Clean all registers
+  mv  x1, zero
+  mv  x2, zero
+  mv  x3, zero
+  mv  x4, zero
+  mv  x5, zero
+  mv  x6, zero
+  mv  x7, zero
+  mv  x8, zero
+  mv  x9, zero
+  mv x10, zero
+  mv x11, zero
+  mv x12, zero
+  mv x13, zero
+  mv x14, zero
+  mv x15, zero
+  mv x16, zero
+  mv x17, zero
+  mv x18, zero
+  mv x19, zero
+  mv x20, zero
+  mv x21, zero
+  mv x22, zero
+  mv x23, zero
+  mv x24, zero
+  mv x25, zero
+  mv x26, zero
+  mv x27, zero
+  mv x28, zero
+  mv x29, zero
+  mv x30, zero
+  mv x31, zero
+
+  # Initialize the stack
   la   sp, _stack_start
+
+_default_handler:
+  j _default_handler
+
+.section .text.start
   
 _start:
   .global _start
 
-  /* clear BSS */
-  la x26, _bss_start
-  la x27, _bss_end
-
-  bge x26, x27, zero_loop_end
-
-zero_loop:
-  sw x0, 0(x26)
-  addi x26, x26, 4
-  ble x26, x27, zero_loop
-zero_loop_end:
-
-/* jump to main program entry point (argc = argv = 0) */
+  # jump to main program entry point (argc = argv = 0) 
   addi x10, x0, 0
```

Comment:
let's have `mv`s here, just as above

and also use mnemonics for `x10` (`a0`) and `x11` (`a1`)